### PR TITLE
Week 4 homework

### PR DESCRIPTION
**Pull request by AlexeyBrambalov**

### CODE REVIEW

File: week-4/Homework/mandatory/2-writers.js [hunk 1]
```diff
@@ -39,6 +39,10 @@ let writers = [
   }
 ];
 
+writers.filter(writer => writer.alive).forEach(writer => console.log(
+ `Hi, my name is ${writer.firstName} ${writer.lastName}. I am ${writer.age} years old, and work as a ${writer.occupation}.`));
+
```
Comment (byohannes) [inline week-4/Homework/mandatory/2-writers.js]: this code does only console out the writers who are alive in the formatted sentence,  but the homework question is asking you to do a different task:
1-Loop through the array, 
and for each object, `console.log()` out the sentence:

"Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
2- and the extra challenge is to console out the writers that are alive.

File: week-4/Homework/mandatory/2-writers.js [hunk 2]
```diff
@@ -39,6 +39,13 @@ let writers = [
...
 
+writers.forEach(writer => console.log(
+ `Hi, my name is ${writer.firstName} ${writer.lastName}. I am ${writer.age} years old, and work as a ${writer.occupation}.`));
+
+
 /*
 If you want an extra challenge, only `console.log()` the writers that are alive.
 */
+
+console.log(writers.filter(writer => writer.alive).map(writer => `${writer.firstName} ${writer.lastName}`));
```
Comment (dahfool) [inline week-4/Homework/mandatory/2-writers.js]: The `map` not necessarily needed as `console.log(writers.filter(writer => writer.alive))` should be enough.

File: week-4/Homework/mandatory/3-water-bottle.js [hunk 3]
```diff
@@ -12,12 +12,15 @@ let bottle = {
...
     // calling this function should make you bottles volume = 100;
+    this.volume = 100;
   },
   drink: function() {
     // calling this function should decrease your bottles volume by 10;
+    this.volume -= 10;
   },
   empty: function() {
     // this function should return true if your bottles volume = 0
+    this.volume === 0 ? true : false;
```
Comment (dahfool) [inline week-4/Homework/mandatory/3-water-bottle.js]: 👍 ternary operator.

File: week-4/InClass/E-arrays-of-objects/exercise-2.js [hunk 4]
```diff
@@ -40,11 +40,21 @@ WRITE YOUR CODE BELOW
 */
 
 
-let destinationNamesWithin500Kms = // Complete here
+let destinationNamesWithin500Kms = travelDestinations.filter(destination => destination.distanceKms < 500).map(distination => distination.destinationName)
```
Comment (dahfool) [inline week-4/InClass/E-arrays-of-objects/exercise-2.js]: `<=` rather than `<` in case `distanceKms` is 500

**Pull request closed without merging**